packages/@aws-cdk/aws-gamelift/lib/game-server-group.ts

```diff
@@ -0,0 +1,563 @@
+    */
+  public readonly role: iam.IRole;
+
+  /**
+    * The principal this GameLift game server group is using.
+    */
+  public readonly grantPrincipal: iam.IPrincipal;
+
+  /**
+   * The VPC network to place the game server group in.
+   */
+  public readonly vpc: ec2.IVpc;
+
+  /**
+    * The game server group's subnets.
+    */
+  public readonly vpcSubnets?: ec2.SubnetSelection;
+
+  /**
+   * The set of Amazon EC2 instance types that GameLift FleetIQ can use when balancing and automatically scaling instances in the corresponding Auto Scaling group.
+   */
+  private readonly instanceDefinitions: InstanceDefinition[] = [];
+
+  constructor(scope: Construct, id: string, props: GameServerGroupProps) {
+    super(scope, id, {
+      physicalName: props.gameServerGroupName,
+    });
+
+    if (!cdk.Token.isUnresolved(props.gameServerGroupName)) {
+      if (props.gameServerGroupName.length > 128) {
+        throw new Error(`GameServerGroup name can not be longer than 128 characters but has ${props.gameServerGroupName.length} characters.`);
+      }
+
+      if (!/^[a-zA-Z0-9-\.]+$/.test(props.gameServerGroupName)) {
+        throw new Error(`Game server group name ${props.gameServerGroupName} can contain only letters, numbers, hyphens, back slash or dot with no spaces.`);
+      }
+    }
+
+    this.vpc = props.vpc;
+    this.vpcSubnets = props.vpcSubnets ?? { subnetType: ec2.SubnetType.PUBLIC };
+    const { subnetIds } = props.vpc.selectSubnets(this.vpcSubnets);
+
+    if (subnetIds.length > 20) {
+      throw new Error(`No more than 20 subnets are allowed per game server group, given ${subnetIds.length}`);
+    }
+
+    // Add all instance definitions
+    if (props.instanceDefinitions.length > 20) {
+      throw new Error(`No more than 20 instance definitions are allowed per game server group, given ${props.instanceDefinitions.length}`);
+    }
+    (props.instanceDefinitions || []).forEach(this.addInternalInstanceDefinition.bind(this));
+
+    this.role = props.role ?? new iam.Role(this, 'ServiceRole', {
+      assumedBy: new iam.CompositePrincipal(
+        new iam.ServicePrincipal('gamelift.amazonaws.com'),
+        new iam.ServicePrincipal('autoscaling.amazonaws.com')),
+      managedPolicies: [
+        iam.ManagedPolicy.fromAwsManagedPolicyName('GameLiftGameServerGroupPolicy'),
+      ],
+    });
+    this.grantPrincipal = this.role;
+
+    const resource = new CfnGameServerGroup(this, 'Resource', {
+      gameServerGroupName: this.physicalName,
+      autoScalingPolicy: this.parseAutoScalingPolicy(props),
+      deleteOption: props.deleteOption,
+      balancingStrategy: props.balancingStrategy,
+      gameServerProtectionPolicy: props.protectGameServer ? 'FULL_PROTECTION' : 'NO_PROTECTION',
+      instanceDefinitions: cdk.Lazy.any({ produce: () => this.parseInstanceDefinitions() }),
+      launchTemplate: this.parseLaunchTemplate(props),
+      minSize: props.minSize,
+      maxSize: props.maxSize,
```

Comment:
These two properties have a minimum value (0 and 1, respectively). We should add a validation for this.

Reply:
ok will do